packages/core/src/components/dialog/dialog.tsx

```diff
@@ -122,6 +126,13 @@ export class Dialog extends AbstractComponent<IDialogProps, {}> {
             </div>
         );
     }
+
+    private handleContainerClick = (evt: React.MouseEvent<HTMLDivElement>) => {
+        // quick re-implementation of canOutsideClickClose because .pt-dialog-container covers the backdrop
+        if (this.props.canOutsideClickClose && evt.currentTarget.closest(`.${Classes.DIALOG}`) == null) {
```

Comment:
I think the second part of the boolean condition here will always be true because `currentTarget` is `.pt-dialog-container`... I think you want `evt.target` instead